doc/sphinx/src/using.rst

```diff
@@ -128,6 +128,65 @@ Please note that none of these functions are thread or MPI aware. In a parallel
 
 as appropriate.
 
+robust_utils.hpp
+^^^^^^^^^^^^^^^^^^^
+
+``robust_utils.hpp`` contains small utility functions for numerical
+robustness, especially around floating point numbers. The available
+functionality is contained in the namespace ``PortsOfCall::Robust`` and includes:
+
+* ``constexpr auto SMALL<T>()`` returns a small number of type ``T``.
+* ``constexpr auto EPS<T>()`` returns a value of type ``T`` close to machine epsilon.
+* ``constexpr auto min_exp_arg<T>()`` returns the smallest of type ``T`` safe value to pass into an exponent.
+* ``constexpr auto max_exp_exp_arg<T>()`` returns the max value of type ``T`` to pass into an exponent.
+* ``auto make_positive(const T val)`` makes the argument of type ``T`` positive.
+
+where here all functionality is templated on type ``T`` and marked
+with ``PORTABLE_INLINE_FUNCTION``. The default type ``T`` is always
+``Real``.
+
+The function
+
+.. code-block:: cpp
+
+  template <typename T>
+  PORTABLE_FORCEINLINE_FUNCTION
+  Real make_bounded(const T val, const T vmin, const T vmax);
+
+bounds ``val`` between ``vmin`` and ``vmax``.
+
+The function
+
+.. code-block:: cpp
+
+  template <typename T>
+  PORTABLE_FORCEINLINE_FUNCTION int sgn(const T &val);
+
+returns the sign of a quantity.
+
+The function
+
+.. code-block:: cpp
+
+  template <typename A, typename B>
+  PORTABLE_FORCEINLINE_FUNCTION auto ratio(const A &a, const B &b)
+
+computes the ratio :math:`A/B` but in a way robust to divide by zero
```

Comment:
See my comment on the code (it doesn't universally avoid division by zero), but also you should define what happens when you compute A / 0 -- Does it give infinity?  Does it give `std::numeric_limits::max`?  Does it use `small()` or `eps()` for B?

Reply:
As we discussed on mattermost, I made it a bit more robust with a masking operation. I also documented what happens when you have (finite)/0.